src/components/Header.js

```diff
@@ -1,22 +1,59 @@
-import React from 'react';
+import React, { Component } from 'react';
 import Button from './Button';
 import { Link } from 'react-router-dom';
 
-const Header = () => {
-  return (
-    <header className="bb br1 b--lavender pa3 bg-white f4 tl serif i">
-      <h1 className="w-60 f3 center ma0 mb2 normal">
-        <Link to="/" className="black no-underline">
-          Call for Diversity
-        </Link>
-        <Link to="/submit-conference">
-          <div className="fr">
-            <Button title="Submit Event" />
-          </div>
-        </Link>
-      </h1>
-    </header>
-  );
-};
+class Header extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isAppOnForm: false
+    };
+    this.handleButtonClick = this.handleButtonClick.bind(this);
+    this.handleBackButtonClick = this.handleBackButtonClick.bind(this);
+  }
+  handleButtonClick() {
+    this.setState({ isAppOnForm: true });
+  }
+  handleBackButtonClick() {
+    this.setState({ isAppOnForm: false });
+  }
+  render() {
+    const submitEventButtonVisibility = this.state.isAppOnForm
+      ? { display: 'none' }
```

Comment:
Have you considered conditionally rendering the `Link` components instead changing the display style?